tuta-sdk/rust/sdk/src/simple_crypto.rs

```diff
@@ -85,3 +90,56 @@ impl From<kyber::KyberEncapsulation> for KyberEncapsulation {
     }
 }
 
+/// Error occurred from trying to encrypt/decrypt with RSA with the simple_crypto
+/// rsa_encrypt/rsa_decrypt methods.
+#[derive(thiserror::Error, Debug, uniffi::Error)]
+#[uniffi(flat_error)]
+enum RSAError {
+    #[error("InvalidKey {reason}")]
+    InvalidKey {
+        reason: String,
+    },
+    #[error("InvalidCiphertext")]
+    InvalidCiphertextError,
+    #[error("RSAEncryptionError {reason}")]
+    RSAEncryptionError {
+        reason: String,
+    },
+}
+
+/// Decrypt with RSA with the given private key components.
+#[uniffi::export]
+fn rsa_decrypt_with_private_key_components(ciphertext: Vec<u8>, modulus: String, private_exponent: String, prime_p: String, prime_q: String) -> Result<Vec<u8>, RSAError> {
+    let modulus = Zeroizing::new(modulus);
+    let private_exponent = Zeroizing::new(private_exponent);
+    let prime_p = Zeroizing::new(prime_p);
+    let prime_q = Zeroizing::new(prime_q);
+
+    let modulus = Zeroizing::new(BASE64_STANDARD.decode(modulus).map_err(|_| RSAError::InvalidKey { reason: "modulus is not valid base64".to_owned() })?);
```

Comment:
I feel like all the base64 should be done outside but also this is how RSA keys always are so maybe it's fine?

Reply:
I assumed we store them like this but turns out it's really us??
https://github.com/tutao/tutanota/blob/016286b102f228944b93f0f711cd62465de37adb/packages/tutanota-crypto/lib/encryption/Rsa.ts#L341

whyyy

we should get rid of it at some point